sw/device/lib/dif/BUILD

```diff
@@ -0,0 +1,658 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+    name = "base",
+    hdrs = [
+        "dif_base.h",
+    ],
+)
+
+cc_library(
+    name = "clkmgr",
+    srcs = [
+        "dif_clkmgr.c",
+    ],
+    hdrs = ["dif_clkmgr.h"],
+    deps = [
+        "//hw/top_earlgrey/ip/clkmgr/data/autogen:clkmgr_regs",
+        "//sw/device/lib/base",
+    ],
+)
+
+cc_test(
+    name = "clkmgr_unittest",
+    srcs = [
+        "dif_clkmgr.c",
+        "dif_clkmgr.h",
+        "dif_clkmgr_unittest.cc",
+    ],
+    defines = [
+        "MOCK_MMIO=1",
+    ],
+    deps = [
+        "//hw/top_earlgrey/ip/clkmgr/data/autogen:clkmgr_regs",
+        "//sw/device/lib/base",
+        "//sw/device/lib/base/testing",
+        "@googletest//:gtest_main",
+    ],
+)
+
+csrng_srcs = ["dif_csrng.c"]
+
+csrng_hdrs = ["dif_csrng.h"]
+
+cc_library(
+    name = "csrng",
+    srcs = csrng_srcs,
+    hdrs = csrng_hdrs,
+    deps = [
+        "//hw/ip/csrng/data:csrng_regs",
+        "//sw/device/lib/base",
+    ],
+)
+
+cc_test(
+    name = "csrng_unittest",
+    srcs = csrng_srcs + csrng_hdrs + [
+        "dif_csrng_unittest.cc",
+    ],
+    defines = [
+        "MOCK_MMIO=1",
+    ],
+    deps = [
+        "//hw/ip/csrng/data:csrng_regs",
+        "//sw/device/lib/base",
+        "//sw/device/lib/base/testing",
+        "@googletest//:gtest_main",
+    ],
+)
+
+edn_srcs = ["dif_edn.c"]
+
+edn_hdrs = ["dif_edn.h"]
+
+cc_library(
+    name = "edn",
+    srcs = edn_srcs,
+    hdrs = edn_hdrs,
+    deps = [
+        "//hw/ip/edn/data:edn_regs",
+        "//sw/device/lib/base",
+    ],
+)
+
+cc_library(
+    name = "uart",
+    srcs = [
+        "autogen/dif_uart_autogen.c",
+        "autogen/dif_uart_autogen.inc",
+        "dif_uart.c",
+    ],
```

Comment:
The listing of srcs & hdrs here is inconsistent with the rest of the file where srcs/hdrs are listed in variables and then those vars are used in the cc_libarary and cc_test rules.

I don't have a strong preference on which is better, but the definitions should be consistent within this BUILD file.

Reply:
They were vestigial. Most of the time having the variables doesn't make sense so I'll continue removing them.